### PR TITLE
Absolute paths (fixes #836)

### LINKED ISSUE #836: CLI directory options should take absolute path
I was attempting to determine if #443, was still a problem.

I ran letsencrypt --work-dir ../Downloads -vvvv run

I was left with the following error...
```
AH00526: Syntax error on line 10 of /etc/apache2/le_dvsni_cert_challenge.conf:
SSLCertificateFile: file '/etc/Downloads/OwYVJOOHVxWaHWNpNUgNDy4TIafggji3RdP0SymKozo.crt' does not exist or is empty
Action 'configtest' failed.
The Apache error log may have more information. 
```

I thought it to be a peculiar place to be looking for a cert.

The config file is populated with...
```
def get_cert_path(self, achall):
        """Returns standardized name for challenge certificate.
        :param achall: Annotated DVSNI challenge.
        :type achall: :class:`letsencrypt.achallenges.DVSNI`
        :returns: certificate file name
        :rtype: str
        """
        return os.path.join(self.configurator.config.work_dir,
                            achall.chall.encode("token") + ".crt")
```

Thus, leading me to the conclusion that we do not handle our directories appropriately from the command-line.  I think it would be much easier if all of the work_dir, config_dir, and logs_dir were guaranteed to be an absolute path.

Providing the full path off the command-line did solve the problem.

Can we find a place to normalize all paths?

### PR DESCRIPTION
**Pull request by bmw**
After letting #1167 sit while I worked on more pressing tasks, I realized that I don't really like that code :/

Here's my 2nd try.

### CODE REVIEW

**Comment by pde** [comment]
LGTM.

File: letsencrypt/cli.py [hunk 1]
```diff
@@ -569,7 +569,7 @@ def plugins_cmd(args, config, plugins):  # TODO: Use IDisplay rather than print
 def read_file(filename, mode="rb"):
     """Returns the given file's contents.
 
-    :param str filename: Filename
+    :param str filename: filename as an absolute path
```
Comment (kuba) [inline letsencrypt/cli.py]: input doesn't have to be absolute, @bmw

File: letsencrypt/configuration.py [hunk 2]
```diff
@@ -37,6 +37,11 @@ class NamespaceConfig(object):
 
     def __init__(self, namespace):
         self.namespace = namespace
+
+        self.namespace.config_dir = os.path.abspath(self.namespace.config_dir)
```
Comment (kuba) [inline letsencrypt/configuration.py]: i liked previous solution better :(

Comment (bmw) [inline letsencrypt/configuration.py]: Really? Why's that?

Sure, the previous solution simplified some of the code in `cli.py`, but that was outside the scope of #836. I think the changes to `configuration.py` are a strict improvement on the old PR.

Previously, I added properties for `cert_path`, `key_path`, `chain_path`, and `fullchain_path`. These should not be properties of the configuration and they are correctly never used that way in the code.

With those changes removed, the only paths that need to be made absolute in `configuration.py` are `config_dir`, `work_dir`, and `logs_dir`. These values do not change during runtime so converting them to an absolute path on every access is unnecessary. Instead, just convert them to an absolute path upfront and all paths based on them will work without further changes.

**Pull request merged**